lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1229,12 +1237,43 @@ def _on_logging_relation_changed(self, event: HookEvent):
                 charm must update its relation data.
         """
         if isinstance(event, RelationEvent):
+            self._set_endpoint_on_relation(event.relation)
             self._process_logging_relation_changed(event.relation)
         else:
             # Upgrade event or other charm-level event
             for relation in self._charm.model.relations[self._relation_name]:
+                self._set_endpoint_on_relation(relation)
                 self._process_logging_relation_changed(relation)
 
+    def _on_logging_relation_departed(self, event: RelationDepartedEvent):
+        """Removes alert rules files when consumer charms left the relation with Loki.
+
+        Args:
+            event: a `CharmEvent` in response to which the Loki
+                charm must update its relation data.
+        """
+        # self._set_endpoint_on_relation(event.relation)
+        self._process_logging_relation_changed(event.relation)
+
+    def update_endpoint(self, relation: Relation = None) -> None:
```

Comment:
To minimize user involvement, would it make sense to have the constructor take an optional arg for extra events to register this handler to?